vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ReferenceManager.java

```diff
@@ -0,0 +1,136 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.vspace.core.data.ReferenceRepository;
+import edu.asu.diging.vspace.core.exception.ReferenceDoesNotExistException;
+import edu.asu.diging.vspace.core.model.IBiblioBlock;
+import edu.asu.diging.vspace.core.model.IReference;
+import edu.asu.diging.vspace.core.model.SortByField;
+import edu.asu.diging.vspace.core.model.impl.BiblioBlock;
+import edu.asu.diging.vspace.core.model.impl.Reference;
+import edu.asu.diging.vspace.core.services.IReferenceManager;
+
+@Service
+public class ReferenceManager implements IReferenceManager {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    
+    @Autowired
+    private ReferenceRepository referenceRepo;
+    
+    @Value("${page_size}")
```

Comment:
does this work or does it need a property source annotation?

Reply:
This works for the pagination .